### PR TITLE
Add even more tasks: token classification, translation and zero shot classification

### PR DESCRIPTION
**Pull request by Wauplin**
Also related to some updates in the specs: https://github.com/huggingface/huggingface.js/pull/872. Related PR is not merged yet but we can still merge this one IMO.

+ everything from https://github.com/huggingface/hub-docs/pull/1399

### CODE REVIEW

File: docs/api-inference/tasks/summarization.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ This is only a subset of the supported models. Find the model that suits you bes
 
 | Payload |  |  |
 | :--- | :--- | :--- |
+| **inputs*** | _string_ | The text to summarize. |
```
Comment (osanseviero) [inline docs/api-inference/tasks/summarization.md:24]: This is missing all the parameters of `text2text-generation` no?

File: docs/api-inference/_toctree.yml [hunk 2]
```diff
@@ -30,5 +30,11 @@
       title: Text Generation
     - local: tasks/text_to_image
       title: Text to Image
+    - local: tasks/token_classification
```
Comment (osanseviero) [inline docs/api-inference/_toctree.yml:33]: I'm wondering if we should add an extra layer to split by modality, but not needed I think?

File: docs/api-inference/tasks/token_classification.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+| **&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(#1)** | _&#x27;none&#x27;_ | Do not aggregate tokens |
+| **&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(#2)** | _&#x27;simple&#x27;_ | Group consecutive tokens with the same label in a single entity. |
+| **&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(#3)** | _&#x27;first&#x27;_ | Similar to "simple", also preserves word integrity (use the label predicted for the first token in a word). |
+| **&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(#4)** | _&#x27;average&#x27;_ | Similar to "simple", also preserves word integrity (uses the label with the highest score, averaged across the word's tokens). |
+| **&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(#5)** | _&#x27;max&#x27;_ | Similar to "simple", also preserves word integrity (uses the label with the highest score across the word's tokens). |
```
Comment (osanseviero) [inline docs/api-inference/tasks/token_classification.md:33]: Note for myself: many of these are not supported in `-community`

File: docs/api-inference/tasks/token_classification.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+<inferencesnippet>
+
+<curl>
+```bash
+curl https://api-inference.huggingface.co/models/dslim/bert-base-NER \
+	-X POST \
+	-d '{"inputs": "My name is Sarah Jessica Parker but you can call me Jessica"}' \
+	-H 'Content-Type: application/json' \
+	-H "Authorization: Bearer hf_***"
+
```
Comment (osanseviero) [inline docs/api-inference/tasks/token_classification.md:78]: Nit I just saw: let's remove the extra newline

File: docs/api-inference/tasks/translation.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+### API specification
+
+#### Request
+
+| Payload |  |  |
+| :--- | :--- | :--- |
+| **inputs*** | _string_ | The text to translate. |
+| **parameters** | _object_ | Additional inference parameters for Translation |
+| **&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;src_lang** | _string_ | The source language of the text. Required for models that can translate from multiple languages. |
+| **&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;tgt_lang** | _string_ | Target language to translate to. Required for models that can translate to multiple languages. |
```
Comment (osanseviero) [inline docs/api-inference/tasks/translation.md:27]: Where do these come from? I'm looking at https://github.com/huggingface/huggingface.js/blob/main/packages/tasks/src/tasks/translation/inference.ts#L20 and https://github.com/huggingface/huggingface.js/blob/main/packages/tasks/src/tasks/translation/spec/input.json

File: docs/api-inference/tasks/zero_shot_classification.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+### API specification
+
+#### Request
+
+| Payload |  |  |
+| :--- | :--- | :--- |
+| **inputs*** | _object_ | The input text data, with candidate labels |
+| **&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;text*** | _string_ | The text to classify |
+| **&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;candidateLabels*** | _string[]_ | The set of possible class labels to classify the text into. |
```
Comment (osanseviero) [inline docs/api-inference/tasks/zero_shot_classification.md:25]: note for Omar, in the current API this is managed through parameters, not inputs?

**Review by osanseviero — COMMENTED** [review]
Cool stuff!

File: docs/api-inference/tasks/summarization.md (hunk 1 above)
Comment (Wauplin) [reply docs/api-inference/tasks/summarization.md]: Yes indeed. I've made some updates in https://github.com/huggingface/huggingface.js/pull/872 and it's better now. Parameters have been added in https://github.com/huggingface/hub-docs/pull/1398/commits/825571b3d9610278cc72359d004ddef0bd865987

File: docs/api-inference/_toctree.yml (hunk 2 above)
Comment (Wauplin) [reply docs/api-inference/_toctree.yml:33]: I wondered as well. For now I'm listing everything by alphabetical order. I think it's fine to keep it like this but if you are opinionated we can keep the same "per-modality split" as in hf.co/tasks.

Comment (Wauplin) [reply docs/api-inference/_toctree.yml:33]: (I'm always using ctrl+F so I'm not of good advice for this :see_no_evil:)

File: docs/api-inference/tasks/translation.md (hunk 5 above)
Comment (Wauplin) [reply docs/api-inference/tasks/translation.md:27]: I opened a PR in `@huggingface.js/tasks` to add modifications to the existing specs (see https://github.com/huggingface/huggingface.js/pull/872). And [here](https://github.com/huggingface/hub-docs/blob/4039c7eb276f14139e86d851259074e9631ad11e/scripts/api-inference/scripts/generate.ts#L18) in the `generate.ts` script I make sure to pull the specs from the `"update-specification-for-docs"` revision. We will have to switch to main once all specs are good.

**Comment by Wauplin** [comment]
Thanks for the review @osanseviero! I've addressed all comments and fixed what had to be fixed. Mind doing another quick pass? :pray:

File: docs/api-inference/_toctree.yml (hunk 2 above)
Comment (osanseviero) [reply docs/api-inference/_toctree.yml:33]: Yeah let's wait for final review from others!

File: docs/api-inference/tasks/translation.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+			},
+			method: "POST",
+			body: JSON.stringify(data),
+		}
+	);
+	const result = await response.json();
+	return result;
+}
+
+query({"inputs": "Меня зовут Вольфганг и я живу в Берлине"}).then((response) => {
```
Comment (osanseviero) [inline docs/api-inference/tasks/translation.md:103]: Maybe an example in english given the model is english->french?

**Review by osanseviero — APPROVED** [review]
Thanks!

**Pull request merged**